pkg/eks/eks.go

```diff
@@ -66,6 +68,36 @@ func (c *ClusterProvider) RefreshClusterStatus(spec *api.ClusterConfig) error {
 	}
 }
 
+var (
+	platformVersionRegex = regexp.MustCompile(`^eks\.(\d+)`)
+)
+
+// SupportsManagedNodes reports whether the cluster supports Managed Nodes
+// The minimum required platform version for Managed Nodes is `eks.3`
+func (c *ClusterProvider) SupportsManagedNodes(clusterConfig *api.ClusterConfig) (bool, error) {
+	if err := c.maybeRefreshClusterStatus(clusterConfig); err != nil {
+		return false, err
+	}
+
+	if platformVersion := c.Status.clusterInfo.cluster.PlatformVersion; platformVersion != nil {
+		match := platformVersionRegex.FindStringSubmatch(*platformVersion)
```

Comment:
Would it be possible to extract the version's parsing logic to an easily unit-testable utility function and to add unit tests for that? (It is easy for subtle bugs to appear and hide in such logic.)

Reply:
Good point, done.

Reply:
Thanks for adding the tests! 👍 ✨ 
What I meant was also to extract such logic to a dedicated function so only `platformVersion string` gets passed, and unit tests are even simpler, but I'll leave that up to you:
```go
	match := platformVersionRegex.FindStringSubmatch(*cluster.PlatformVersion)
	if len(match) != 2 {
		return false, fmt.Errorf("failed to parse cluster's platform version: %q", *cluster.PlatformVersion)
	}
	versionStr := match[1]
	minSupportedVersion := 3
	version, err := strconv.Atoi(versionStr)
	if err != nil {
		return false, err
	}
	return version >= minSupportedVersion, nil
```

Reply:
Both Kubernetes version and platform version are required to determine whether managed nodes are supported or not, so IMO, not much is gained by separating them into two functions. This way, the tests execute the same function as the actual code.